src/mmw/mmw/settings/base.py

```diff
@@ -244,6 +244,7 @@ def get_env_setting(setting):
     'django.contrib.sessions.middleware.SessionMiddleware',
     'django.middleware.csrf.CsrfViewMiddleware',
     'django.contrib.auth.middleware.AuthenticationMiddleware',
+    'apps.user.middleware.ItsiAuthenticationMiddleware',
```

Comment:
If a middleware returns a redirect, then subsequent middlewares will not run, according to the Django docs. So it's safer to put it at the end of the list.